x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -54,35 +59,116 @@ export const enhancedEsSearchStrategyProvider: TSearchStrategyProvider<typeof ES
   return { search, cancel };
 };
 
+async function getBackgroundSession(
+  request: IEnhancedEsSearchRequest,
+  options?: ISearchOptions,
+  context?: IEnhancedSearchContext
+) {
+  if (
+    context &&
+    context.backgroundSearchService &&
+    options?.rawRequest &&
+    !!request.sessionId &&
+    !request.id
+  ) {
+    return await context.backgroundSearchService.getId(
+      options?.rawRequest,
+      request.sessionId,
+      request.params.body
+    );
+  }
+}
+
+function trackBackgroundSearch(
+  request: IEnhancedEsSearchRequest,
+  asyncId: string,
+  options?: ISearchOptions,
+  context?: IEnhancedSearchContext
+) {
+  if (
+    context &&
+    context.backgroundSearchService &&
+    options?.rawRequest &&
+    !!request.sessionId &&
+    !!asyncId &&
+    !request.id
```

Comment:
nit: Both of the `if` statements in `trackBackgroundSearch` and `getBackgroundSession` are a bit tricky to reason about; perhaps they could be simplified somehow

For example if `asyncId` is a required param & a string, why do we need to check `!!asyncId` here?